vignettes/advanced_features.Rmd

```diff
@@ -0,0 +1,108 @@
+---
+title: "Advanced Features"
+author: "Marcus Young, modified by Malcolm Morganr"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+The vignette introduces some of the more advanced features of OTP, and gives some examples of the types of anlaysis that are possible when using OTP and R togther.
+
+### Recap
+
+For this vingnette we will use the same data as the `getting_started` vingette. If you have not yet created the example graph you can set it up with the following commands. If you are using non-default settings see the `getting_started` vingette for full details.
+
+```{r eval = FALSE}
+library(opentripplanner)
+# Path to a folder containing the OTP.jar file, change to where you saved the file.
+path_otp <- "C:/otp"
+
+path_data <- system.file("extdata", package="opentripplanner")
+log <- otp_build_graph(otp = path_otp, dir = path_data)
+otp_setup(otp = path_otp, dir = path_data, memory = 1.5)
+otpcon <- otp_connect()
+
+```
+
+## Isochrones
+
+Now that we have a working instance of OTP, we'll use it to generate some travel-time isochrones. We are interested in visualising how long it takes to access Manchester Airport using public transport from different parts of Greater Manchester. We will do this by requesting isochrones from OTP for 15, 30, 45, 60, 75 and 90 minutes. This can be achieved with a single function `otp_isochrone()`.
+
+```{r eval = FALSE}
+airport_current  <- otp_isochrone(otpcon = otpcon,
+            fromPlace = c(53.36367, -2.27190), # latlong of Manchester Airport
+            mode = c("WALK","TRANSIT"),
+            cutoffSec = c(15, 30, 45, 60, 75, 90) * 60 ) # Cut offs in seconds
+airport_current$minutes = airport_current$time / 60 # Convert back to minutes
+
+```
+
+We can visualise the isochrones on a map using the `tmap` package.
+
+```{r, eval = FALSE}
+library(tmap)                       # Load the tmap package
+tmap_mode("view")                   # Set tmap to interative viewing
+map <- tm_shape(airport_current) +  # Build the map
+  tm_fill("minutes",
+          breaks = c(0, 15.001, 30.001, 45.001, 60.001, 75.001, 90.001),
+          style = "fixed",
+          palette = "-RdYlBu") +
+  tm_borders()
+map                                 # Plot the map
+```
+
+You should see a map like this.
+
+```{r, echo = FALSE, fig.align='center', fig.cap="\\label{fig:otpgui}Isochrones from Manchester Airport"}
+knitr::include_graphics("images/isochrone.png")
+```
```

Comment:
@mem48 There's something not right about these isochrones in the map. Full of holes. Compare with the one in the tutorial. It may due to WALK being included in the modes. Isochrones only work correctly for TRANSIT modes.

Reply:
Actually, given the other issue regarding date format mention above, I suspect this is actually the WALK isochrone for Manchester Airport as no Transit would have been returned. But WALK won't returrn a proper valid isochrone, thus all the holes.